### ISSUE TITLE
Can't Create Projects or View To-Dos

### ISSUE THREAD

**Issue by danieltogra** [TracksApp/tracks#2023]
Once I create a recurring To-do, I click on Home and it displays an error to check the log files. After that, everywhere I click I get the same error again. i have to flush the database for it to work again. This is what it shows:
```
I, [2016-06-21T16:05:43.608730 #6967]  INFO -- :   Rendered todos/_todo.html.erb (13.6ms)
I, [2016-06-21T16:05:43.608848 #6967]  INFO -- :   Rendered todos/_collection.html.erb (14.5ms)
I, [2016-06-21T16:05:43.608948 #6967]  INFO -- :   Rendered contexts/_context.html.erb (15.6ms)
I, [2016-06-21T16:05:43.609065 #6967]  INFO -- :   Rendered contexts/show.html.erb within layouts/application (16.4ms)
I, [2016-06-21T16:05:43.609240 #6967]  INFO -- : Completed 500 Internal Server Error in 55ms (ActiveRecord: 6.2ms)
F, [2016-06-21T16:05:43.613117 #6967] FATAL -- : 
ActionView::Template::Error (File name too long - /home/daniel/tracks-2.3-4.0/tmp/cache/views%2Ftodos%2F3-20160621210538000000000%2F20160621%2Fcontext%2Ffalse%2Fc9a33e74913927d9ce9c62c2eadea771%2Fc9a33e74913927d9ce9c62c2eadea77120160621-6967-xbug51.lock):
    10: # invalidate the cache every day because of staleness or
    11: # rendering of "due in x days" that change without touching updated at of the todo
    12: # also make different caches per source_view to handle difference in showing [C] and [P]
    13: cache [todo, current_user.date.strftime("%Y%m%d"), @source_view, current_user.prefs.verbose_action_descriptors] do
    14: %>
    15:   <div id="<%= dom_id(todo) %>" class="item-container">
    16:     <div id="<%= dom_id(todo, 'line') %>" class="item-show">
  app/views/todos/_todo.html.erb:13:in `_app_views_todos__todo_html_erb__792608391113674462_37256420'
  app/helpers/todos_helper.rb:131:in `block in todos_container_items'
  app/helpers/todos_helper.rb:129:in `todos_container_items'
  app/views/todos/_collection.html.erb:6:in `block in _app_views_todos__collection_html_erb___4293963971803460381_35281900'
  app/helpers/todos_helper.rb:108:in `block in todos_container'
  app/helpers/todos_helper.rb:104:in `todos_container'
  app/views/todos/_collection.html.erb:4:in `_app_views_todos__collection_html_erb___4293963971803460381_35281900'
  app/views/contexts/_context.html.erb:17:in `block in _app_views_contexts__context_html_erb__3914644992945296109_36963620'
  app/views/contexts/_context.html.erb:5:in `_app_views_contexts__context_html_erb__3914644992945296109_36963620'
  app/helpers/todos_helper.rb:30:in `show_grouped_todos'
  app/views/contexts/show.html.erb:11:in `_app_views_contexts_show_html_erb__192591852744742045_40775800'
  app/controllers/contexts_controller.rb:54:in `show'
```

The error says 'File name too long'

Im using Ubuntu 14.04.2
environment:
```
daniel@techcloud:~/tracks-2.3-4.0$ sudo bundle exec rails server -e production
=> Booting WEBrick
=> Rails 4.1.11 application starting in production on http://0.0.0.0:3000
=> Run `rails server -h` for more startup options
=> Notice: server is listening on all interfaces (0.0.0.0). Consider using 127.0.0.1 (--binding option)
=> Ctrl-C to shutdown server
[2016-06-21 12:21:32] INFO  WEBrick 1.3.1
[2016-06-21 12:21:32] INFO  ruby 1.9.3 (2013-11-22) [x86_64-linux]
[2016-06-21 12:21:32] INFO  WEBrick::HTTPServer#start: pid=6967 port=3000
```

**Comment by C-Otto** [comment]
Which file system is that?

**Comment by danieltogra** [comment]
It is the default for Ubuntu. EXT4 @C-Otto

**Comment by dnrce** [comment]
This was fixed in Rails 4.2.0 but not Rails 4.1.x: https://github.com/rails/rails/commit/9010274627a538c688e44737e7a50c659da7e863

As a result, this won't be an issue in Tracks `master` and isn't really fixable in 2.3.0.
